### PR TITLE
Support overriding global routing status of routing policies

### PR DESCRIPTION
**Pull request by mpolden**
Next step is to wire this into application and operator APIs.

FYI @tokle

### CODE REVIEW

**Comment by mpolden** [comment]
Last three commits are the meat and potatoes.

**Review by oyving — APPROVED** [review]
I just had some minor naming nits and a question of collisions in the policy map (but I don't think this was properly handled earlier either).

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/routing/Status.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.Objects;
+
+/**
+ * Represents the status of a routing policy.
+ *
+ * This is immutable.
+ *
+ * @author mpolden
+ */
+public class Status {
```
[the remark this answers is not in the document]
Comment (mpolden) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/routing/Status.java]: It's two things now: Load balancer active status and global routing status.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RoutingPolicySerializer.java [hunk 2]
```diff
@@ -57,22 +59,23 @@ public Slime toSlime(Set<RoutingPolicy> routingPolicies) {
...
-                                           endpointIds,
-                                           inspect.field(loadBalancerActiveField).asBool()));
+            var id = new RoutingPolicyId(owner,
+                                         ClusterSpec.Id.from(inspect.field(clusterField).asString()),
+                                         ZoneId.from(inspect.field(zoneField).asString()));
+            policies.put(id, new RoutingPolicy(id,
+                                               HostName.from(inspect.field(canonicalNameField).asString()),
+                                               Serializers.optionalString(inspect.field(dnsZoneField)),
+                                               endpointIds,
+                                               inspect.field(loadBalancerActiveField).asBool()));
```
[the remark this answers is not in the document]
Comment (mpolden) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RoutingPolicySerializer.java]: The data read here is the result of `toSlime(Map<RoutingPolicyId, RoutingPolicy>)`, so collisions can only occur if `toSlime` somehow writes the same cluster+zone pair more than once or if an external entity messes with the ZK data. The former is covered by `RoutingPolicySerializerTest`.

**Comment by mpolden** [comment]
PTAL.

**Review by oyving — APPROVED** [review]

**Pull request merged**